### ISSUE TITLE
cookie filter using select control is loaded before url data populated control

### ISSUE THREAD

**Issue by albfan** [wenzhixin/bootstrap-table#5105]
**Bootstraptable version(s) affected**:
1.16 and develop

**Description**  

When cookie extension is enabled, filter is a select control and keys do not match values and data is loaded from url, the option is added before options are populated.

relates with #1840

**Example**  

Before: https://live.bootstrap-table.com/code/albfan/3504
After: https://live.bootstrap-table.com/code/albfan/3502

**Possible  (optional)**  

See this solution is more than a hack. It assumes that a select filled with url data has an two options (one empty and other with the cookie value) It gets the cookie value from that second option, and use as selected option.

In the example is barely perceptible, but on a laggy service you can see the option with the value for a while, until it is correctly reloaded.

**Comment by UtechtDustin** [comment]
I cant reproduce the issue on the Before example, there is no new added option on the select element.

**Comment by albfan** [comment]
Yes, sometimes it works for me. Sometimes not. I try to emulate a laggy service here:

https://live.bootstrap-table.com/code/albfan/3517

But still doesn't show all the time.

**Comment by nickv2002** [comment]
We're hitting the same bug.

Extra context: it looks like it's related to this old closed issue that got dropped: https://github.com/wenzhixin/bootstrap-table/issues/1840

Thanks for the work on the fix @albfan

**Comment by albfan** [comment]
I realize the problem might be almost solved on develop branch. (I still see from time to time) Where the code is always failing is using CDN 1.16

https://live.bootstrap-table.com/code/albfan/3579

![url-select-cookie](https://user-images.githubusercontent.com/220968/85292681-91aaa900-b49c-11ea-9749-3a39f38496d1.gif)

As code is called in async way:

https://github.com/wenzhixin/bootstrap-table/blob/develop/src/extensions/filter-control/bootstrap-table-filter-control.js#L544

There's no doubt cookie will fill the select option before ajax returns.

@nickv2002 so I think some fix is already done on develop branch for this.

**Comment by UtechtDustin** [comment]
@albfan no sorry, i also have to do a diff.

**Comment by albfan** [comment]
On version 1.17 it is almost resolved (just can see one time and the it works all the time (not sure how to create an example to see that for first one time)

Will reopen if I found what's happening but we can see this as resolved
